liblttng-ust-java-agent/java/lttng-ust-agent-common/org/lttng/ust/agent/AbstractLttngAgent.java

```diff
@@ -324,31 +280,43 @@ public boolean appContextDisabled(String contextRetrieverName, String contextNam
 
 	@Override
 	public boolean isEventEnabled(String eventName) {
-		/* If at least one session enabled the "*" wildcard, send the event */
-		if (enabledWildcards.get() > 0) {
-			return true;
+		Boolean cachedEnabled = enabledEventNamesCache.get(eventName);
+		if (cachedEnabled != null) {
+			/* We have seen this event previously */
+			/*
+			 * Careful! enabled == null could also mean that the null value is
+			 * associated with the key. But we should have never inserted null
+			 * values in the map.
+			 */
+			return cachedEnabled.booleanValue();
 		}
 
-		/* Check if at least one session wants this exact event name */
-		if (enabledEvents.containsKey(eventName)) {
-			return true;
-		}
+		/*
+		 * We have not previously checked this event. Run it against all known
+		 * enabled event patterns to determine if it should pass or not.
+		 */
+		synchronized (enabledEventNamesCache) {
```

Comment:
is it the first "synchronized" that we add on the tracing path ? Even though it's not used often, I want to ensure that we are allowed to do a synchronized here, and that we don't cause issues with log4j/jul.